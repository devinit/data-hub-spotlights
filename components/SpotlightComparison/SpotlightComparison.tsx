import React, { FunctionComponent } from 'react';
import classNames from 'classnames';

interface ComparisonSelectorFormProps {
  legendClassName?: string;
}

const SpotlightComparison: FunctionComponent<ComparisonSelectorFormProps> = ({ legendClassName, children }) => {
  return (
    <div className="spotlight__comparison">
      <span className={classNames('spotlight__comparison-legend', legendClassName)}></span>
      {children}
    </div>
  );
};

export { SpotlightComparison };
