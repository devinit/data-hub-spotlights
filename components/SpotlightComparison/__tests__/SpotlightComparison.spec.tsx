/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { SpotlightComparison } from '../SpotlightComparison';

describe('SpotlightComparison', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<SpotlightComparison />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders the legend with the configured class', () => {
    const renderer = TestRenderer.create(
      <SpotlightComparison legendClassName="spotlight__comparison-legend--alt" />
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
