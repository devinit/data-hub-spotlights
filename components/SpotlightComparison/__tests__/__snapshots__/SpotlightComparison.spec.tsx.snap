// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpotlightComparison renders correctly 1`] = `
<div
  className="spotlight__comparison"
>
  <span
    className="spotlight__comparison-legend"
  />
</div>
`;

exports[`SpotlightComparison renders the legend with the configured class 1`] = `
<div
  className="spotlight__comparison"
>
  <span
    className="spotlight__comparison-legend spotlight__comparison-legend--alt"
  />
</div>
`;
